docs/integrations/ldap/org.md

```diff
@@ -314,3 +314,66 @@ builder.addProcessor(
   }),
 );
 ```
+
+## Using a Provider instead of a Processor
+
+An alternative to using the Processor for ingesting LDAP entries is to use a
+Provider. Doing this can give you a little bit more freedom to handle the LDAP
+ingestion more independently from the rest of the catalog ingestion.
+
+This can be useful if you have a lot of Users and Groups and hitting your LDAP
+server is resource intensive but you still want your other catalog entries to be
+updated frequently.
+
+> Note: When configuring to use a Provider instead of a Processor you do not
+> need to add a _location_ pointing to your LDAP server
+
+```ts
+// packages/backend/src/plugins/catalog.ts
+import { CatalogBuilder } from '@backstage/plugin-catalog-backend';
+import { PluginEnvironment } from '../types';
+
+import { Router } from 'express';
+import { LdapOrgEntityProvider } from '@backstage/plugin-catalog-backend-module-ldap';
+
+import { TaskScheduler } from '@backstage/backend-tasks';
+import { Duration } from 'luxon';
+
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
+  const scheduler = TaskScheduler.fromConfig(env.config).forPlugin('catalog');
+  const ldapEntityProvider = LdapOrgEntityProvider.fromConfig(env.config, {
+    id: 'custom-ldap',
+    // target needs to match the catalog.processors.ldapOrg.providers.target specified in app-config
+    target: 'ldaps://ds.example.net',
+    logger: env.logger,
+  });
+
+  await scheduler.scheduleTask({
+    id: 'refresh-ldap',
+    // initialDelay is the time to wait before running the first cycle, you might need to adjust this delay to make sure that the ldapEntityProvider has been initialized before running.
+    initialDelay: Duration.fromMillis(10000),
+    // frequency sets how often you want to ingest users and groups from LDAP, in this case every 60 minutes
+    frequency: Duration.fromObject({ minutes: 60 }),
+    timeout: Duration.fromObject({ minutes: 15 }),
+    fn: async () => {
+      try {
+        await ldapEntityProvider.read();
+      } catch (error) {
+        console.error(error);
```

Comment:
Probably should encourage using the `env.logger` here instead of the console. And I realize, the task framework itself should probably be amended do logging in these cases ... I don't think it does now.

Reply:
Yeah, we were thinking of removing the try-catch altogether, but the reason we left it in for these docs is that we ourselves ran into the issue that before we added the initialdelay the data never appeared to load on start. 

Looking into that issue we found that in the read function for LdapOrgEntityProvider you throw when the connection isn't initialized: 
```ts
    if (!this.connection) {
      throw new Error("Not initialized");
    }
```

This error never ended up in our terminal, so without the catching it looked like a silent fail. 

Do you want me to keep the try-catch and change it to use env.logger instead or should I remove the try-catch entirely?

Reply:
You can keep it and change to env.logger I'd say, since we don't log anything at the framework level right now so it's still useful for users to see that, or other, failure modes.

Actually, you may also want to move this scheduling to after both `catalog.build` and `processingEngine.start` below. At that point, all providers will have been connected. Then you won't need to have the initial delay.

Reply:
👍  moved it below the start, and changed to env.logger